app/poros/spells.rb

```diff
@@ -0,0 +1,185 @@
+    @all_spells = Spell.where('classes LIKE ?', "%#{character_class}%")
+    if level == 1
+      find_spells(level, class_table, 'sorcerer_one')
+    else
+      find_spells(level, class_table, 'sorcerer_higher')
+    end
+  end
+
+  def warlock_spells(character_class, level, class_table)
+    @all_spells = Spell.where('classes LIKE ?', "%#{character_class}%")
+    if [11, 13, 15, 17].include?(level)
+      find_spells(level, class_table, 'warlock_adjusted')
+    else
+      find_spells(level, class_table, 'warlock_normal')
+    end
+  end
+
+  def find_spells(level, class_table, tag)
+    class_table.each do |row|
+      if row.scan(/\d+/)[0].to_i == level
+        find_spell_slots(row, tag)
+        find_current_spells
+      end
+    end
+  end
+
+  def find_spell_slots(row, tag)
+    spell_level = 0
+    case tag
+    when 'simple'
+      row.scan(/\d+/)[2..-1].each do |number|
+        @spell_slots[spell_level] = number.to_i
+        spell_level += 1
+      end
+    when 'extra_number'
+      row.scan(/\d+/)[3..-1].flatten.each do |number|
+        @spell_slots[spell_level] = number.to_i
+        spell_level += 1
+      end
+    when 'cleric_five'
+      row.scan(/\d+/)[4..-1].flatten.each do |number|
+        @spell_slots[spell_level] = number.to_i
+        spell_level += 1
+      end
+    when 'paladin'
+      row.scan(/\d+/)[2..-1].flatten.each do |number|
+        spell_level += 1
+        @spell_slots[spell_level] = number.to_i
+      end
+    when 'ranger'
+      if row.scan(/\d+/)[3..-1].nil?
+        @spell_slots = {}
+      else
+        row.scan(/\d+/)[3..-1].flatten.each do |number|
+          spell_level += 1
+          @spell_slots[spell_level] = number.to_i
+        end
+      end
+    when 'sorcerer_one'
+      # Point of fact, this doesn't EXACTLY match the rules, but it's close, and can be refactored in the future
+      [row.scan(/\d+/)[2], row.scan(/\d+/)[4..-1]].flatten.each do |number|
+        @spell_slots[spell_level] = number.to_i
+        spell_level += 1
+      end
+    when 'sorcerer_higher'
+      # Point of fact, this doesn't EXACTLY match the rules, but it's close, and can be refactored in the future
+      [row.scan(/\d+/)[3], row.scan(/\d+/)[5..-1]].flatten.each do |number|
+        @spell_slots[spell_level] = number.to_i
+        spell_level += 1
+      end
+    when 'warlock_normal'
+      # Also not exactly how it works, but this works for now and can be refactored
+      @spell_slots[spell_level] = row.scan(/\d+/)[2].to_i
+      @spell_slots[row.scan(/\d+/)[-2].to_i] = row.scan(/\d+/)[4].to_i
+    when 'warlock_adjusted'
+      @spell_slots[spell_level] = row.scan(/\d+/)[3].to_i
+      @spell_slots[row.scan(/\d+/)[-2].to_i] = row.scan(/\d+/)[5].to_i
+    end
+  end
+
+  def find_current_spells
+    @spell_slots.each do |slot, spells|
+      @current_spells << @all_spells.where(level: slot).sample(spells)
+    end
+  end
+
+  def determine_archetype_spells(level, archetype)
```

Comment:
This Brute Force method here is for archetypes that aren't in the API call, so this is just in case something like this were to come up in the future.